plugins/bcm57xx/fu-bcm57xx-device.c

```diff
@@ -0,0 +1,847 @@
+/*
+ * Copyright (C) 2018-2020 Evan Lojewski
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: GPL-2+
+ */
+
+#include "config.h"
+
+#include "fu-common.h"
+#include "fu-bcm57xx-common.h"
+#include "fu-bcm57xx-device.h"
+#include "fu-bcm57xx-firmware.h"
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <errno.h>
+#ifdef HAVE_MMAN_H
+#include <sys/mman.h>
+#endif
+#ifdef HAVE_ETHTOOL_H
+#include <linux/ethtool.h>
+#include <linux/sockios.h>
+#include <net/if.h>
+#endif
+#ifdef HAVE_IOCTL_H
+#include <sys/ioctl.h>
+#endif
+#ifdef HAVE_SOCKET_H
+#include <sys/socket.h>
+#endif
+
+#ifdef HAVE_VALGRIND
+#include <valgrind.h>
+#endif /* HAVE_VALGRIND */
+
+typedef struct {
+	guint8	*buf;
+	gsize	 bufsz;
+} FuBcm57xxMmap;
+
+#define FU_BCM57XX_BAR_DEVICE	0
+#define FU_BCM57XX_BAR_APE	2
```

Comment:
This should be #define FU_BCM57XX_BAR_APE	1
APE is at resource2, which is in bar[1] since you are multiplying *2 inside of fu_bcm57xx_device_open

Reply:
Indeed, thanks!